// @expo/prebuild-config/build/plugins/unversioned/expo-splash-screen/withIosSplashAssets.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildContentsJsonImages = exports.withIosSplashAssets = void 0;
const config_plugins_1 = require("@expo/config-plugins");
const debug_1 = __importDefault(require("debug"));
const fs_extra_1 = __importDefault(require("fs-extra"));
const es_1 = __importDefault(require("jimp/es"));
const path = __importStar(require("path"));
const AssetContents_1 = require("../../icons/AssetContents");
const debug = debug_1.default('expo:prebuild-config:expo-splash-screen:ios:assets');
const IMAGESET_PATH = 'Images.xcassets/SplashScreen.imageset';
const BACKGROUND_IMAGESET_PATH = 'Images.xcassets/SplashScreenBackground.imageset';
const PNG_FILENAME = 'image.png';
const DARK_PNG_FILENAME = 'dark_image.png';
const TABLET_PNG_FILENAME = 'tablet_image.png';
const DARK_TABLET_PNG_FILENAME = 'dark_tablet_image.png';
const withIosSplashAssets = (config, splash) => {
    if (!splash) {
        return config;
    }
    return config_plugins_1.withDangerousMod(config, [
        'ios',
        async (config) => {
            var _a, _b;
            const projectPath = config_plugins_1.IOSConfig.Paths.getSourceRoot(config.modRequest.projectRoot);
            await createSplashScreenBackgroundImageAsync({
                iosNamedProjectRoot: projectPath,
                splash,
            });
            await configureImageAssets({
                projectPath,
                image: splash.image,
                darkImage: (_a = splash.dark) === null || _a === void 0 ? void 0 : _a.image,
                tabletImage: splash.tabletImage,
                darkTabletImage: (_b = splash.dark) === null || _b === void 0 ? void 0 : _b.tabletImage,
            });
            return config;
        },
    ]);
};
exports.withIosSplashAssets = withIosSplashAssets;
/**
 * Creates imageset containing image for Splash/Launch Screen.
 */
async function configureImageAssets({ projectPath, image, darkImage, tabletImage, darkTabletImage, }) {
    const imageSetPath = path.resolve(projectPath, IMAGESET_PATH);
    // ensure old SplashScreen imageSet is removed
    await fs_extra_1.default.remove(imageSetPath);
    if (!image) {
        return;
    }
    await writeContentsJsonFileAsync({
        assetPath: imageSetPath,
        image: PNG_FILENAME,
        darkImage: darkImage ? DARK_PNG_FILENAME : null,
        tabletImage: tabletImage ? TABLET_PNG_FILENAME : null,
        darkTabletImage: darkTabletImage ? DARK_TABLET_PNG_FILENAME : null,
    });
    await copyImageFiles({ projectPath, image, darkImage, tabletImage, darkTabletImage });
}
async function createPngFileAsync(color, filePath) {
    const png = new es_1.default(1, 1, color);
    return png.writeAsync(filePath);
}
async function createBackgroundImagesAsync({ projectPath, color, darkColor, tabletColor, darkTabletColor, }) {
    await generateImagesAssetsAsync({
        async generateImageAsset(item, fileName) {
            await createPngFileAsync(item, path.resolve(projectPath, BACKGROUND_IMAGESET_PATH, fileName));
        },
        anyItem: color,
        darkItem: darkColor,
        tabletItem: tabletColor,
        darkTabletItem: darkTabletColor,
    });
}
async function copyImageFiles({ projectPath, image, darkImage, tabletImage, darkTabletImage, }) {
    await generateImagesAssetsAsync({
        async generateImageAsset(item, fileName) {
            await fs_extra_1.default.copyFile(item, path.resolve(projectPath, IMAGESET_PATH, fileName));
        },
        anyItem: image,
        darkItem: darkImage,
        tabletItem: tabletImage,
        darkTabletItem: darkTabletImage,
    });
}
async function generateImagesAssetsAsync({ generateImageAsset, anyItem, darkItem, tabletItem, darkTabletItem, }) {
    const items = [
        [anyItem, PNG_FILENAME],
        [darkItem, DARK_PNG_FILENAME],
        [tabletItem, TABLET_PNG_FILENAME],
        [darkTabletItem, DARK_TABLET_PNG_FILENAME],
    ].filter(([item]) => !!item);
    await Promise.all(items.map(([item, fileName]) => generateImageAsset(item, fileName)));
}
async function createSplashScreenBackgroundImageAsync({ iosNamedProjectRoot, splash, }) {
    var _a, _b;
    const color = splash.backgroundColor;
    const darkColor = (_a = splash.dark) === null || _a === void 0 ? void 0 : _a.backgroundColor;
    const tabletColor = splash.tabletBackgroundColor;
    const darkTabletColor = (_b = splash.dark) === null || _b === void 0 ? void 0 : _b.tabletBackgroundColor;
    const imagesetPath = path.join(iosNamedProjectRoot, BACKGROUND_IMAGESET_PATH);
    // Ensure the Images.xcassets/... path exists
    await fs_extra_1.default.remove(imagesetPath);
    await fs_extra_1.default.ensureDir(imagesetPath);
    await createBackgroundImagesAsync({
        projectPath: iosNamedProjectRoot,
        color,
        darkColor: darkColor ? darkColor : null,
        tabletColor: tabletColor ? tabletColor : null,
        darkTabletColor: darkTabletColor ? darkTabletColor : null,
    });
    await writeContentsJsonFileAsync({
        assetPath: path.resolve(iosNamedProjectRoot, BACKGROUND_IMAGESET_PATH),
        image: PNG_FILENAME,
        darkImage: darkColor ? DARK_PNG_FILENAME : null,
        tabletImage: tabletColor ? TABLET_PNG_FILENAME : null,
        darkTabletImage: darkTabletColor ? DARK_TABLET_PNG_FILENAME : null,
    });
}
const darkAppearances = [
    {
        appearance: 'luminosity',
        value: 'dark',
    },
];
function buildContentsJsonImages({ image, darkImage, tabletImage, darkTabletImage, }) {
    return [
        // Phone light
        AssetContents_1.createContentsJsonItem({
            idiom: 'universal',
            filename: image,
            scale: '1x',
        }),
        AssetContents_1.createContentsJsonItem({
            idiom: 'universal',
            scale: '2x',
        }),
        AssetContents_1.createContentsJsonItem({
            idiom: 'universal',
            scale: '3x',
        }),
        // Phone dark
        darkImage &&
            AssetContents_1.createContentsJsonItem({
                idiom: 'universal',
                appearances: darkAppearances,
                filename: darkImage,
                scale: '1x',
            }),
        darkImage &&
            AssetContents_1.createContentsJsonItem({
                idiom: 'universal',
                appearances: darkAppearances,
                scale: '2x',
            }),
        darkImage &&
            AssetContents_1.createContentsJsonItem({
                idiom: 'universal',
                appearances: darkAppearances,
                scale: '3x',
            }),
        // Tablet light
        tabletImage &&
            AssetContents_1.createContentsJsonItem({
                idiom: 'ipad',
                filename: tabletImage,
                scale: '1x',
            }),
        tabletImage &&
            AssetContents_1.createContentsJsonItem({
                idiom: 'ipad',
                scale: '2x',
            }),
        // Phone dark
        darkTabletImage &&
            AssetContents_1.createContentsJsonItem({
                idiom: 'ipad',
                appearances: darkAppearances,
                filename: darkTabletImage !== null && darkTabletImage !== void 0 ? darkTabletImage : undefined,
                scale: '1x',
            }),
        darkTabletImage &&
            AssetContents_1.createContentsJsonItem({
                idiom: 'ipad',
                appearances: darkAppearances,
                scale: '2x',
            }),
    ].filter(Boolean);
}
exports.buildContentsJsonImages = buildContentsJsonImages;
async function writeContentsJsonFileAsync({ assetPath, image, darkImage, tabletImage, darkTabletImage, }) {
    const images = buildContentsJsonImages({ image, darkImage, tabletImage, darkTabletImage });
    debug(`create contents.json:`, assetPath);
    debug(`use images:`, images);
    await AssetContents_1.writeContentsJsonAsync(assetPath, { images });
}
